File: spigot/src/main/java/net/isariel/gaia/spigot/GaiaSpigotPlugin.java
package net.isariel.gaia.spigot;

import net.isariel.gaia.spigot.module.GaiaModule;
import net.isariel.hermes.api.HermesAPI;
import net.isariel.hermes.api.HermesAgent;
import org.bukkit.plugin.java.JavaPlugin;

/**
 * @author Felix Klauke <fklauke@itemis.de>
 */
public class GaiaSpigotPlugin extends JavaPlugin {

    private GaiaApplication application;

    @Override
    public void onDisable() {
        application.doDisable();
    }

    @Override
    public void onEnable() {
        HermesAgent hermesAgent = HermesAPI.getAgent();

        application = hermesAgent.initializeApplication(GaiaApplication.class, new GaiaModule(this));
        application.doEnable();
    }
}

File: spigot/src/main/java/net/isariel/gaia/spigot/GaiaSpigotPlugin.kt
package net.isariel.gaia.spigot

import net.isariel.gaia.spigot.module.GaiaModule
import net.isariel.hermes.api.HermesAPI
import org.bukkit.plugin.java.JavaPlugin

/**
 * @author Felix Klauke <info@felix-klauke.de>
 */
class GaiaSpigotPlugin : JavaPlugin() {

    private lateinit var application: GaiaApplication

    override fun onDisable() {
        application.doDisable()
    }

    override fun onEnable() {
        val hermesAgent = HermesAPI.getAgent()

        application = hermesAgent.initializeApplication(GaiaApplication::class.java, GaiaModule(this))
        application.doEnable()
    }
}
